.github/workflows/static_analysis.yml

```diff
@@ -0,0 +1,85 @@
+# contains tests that can run without compiling LightGBM, like
+# linters and documentation checks.
+name: Static Analysis
+
+on:
+  push:
+    branches:
+    - master
+  pull_request:
+    branches:
+    - master
+
+env:
+  CONDA_ENV: test-env
+  GITHUB_ACTIONS: 'true'
+  PYTHON_VERSION: 3.8
+
+jobs:
+  test:
+    name: ${{ matrix.task }}
+    runs-on: ubuntu-latest
+    timeout-minutes: 60
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - task: lint
+          - task: check-docs
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2.3.4
+        with:
+          fetch-depth: 5
+          submodules: false
+      - name: Setup and run tests
+        shell: bash
+        run: |
+          export TASK="${{ matrix.task }}"
+          export OS_NAME="linux"
+          export BUILD_DIRECTORY="$GITHUB_WORKSPACE"
+          export LGB_VER=$(head -n 1 VERSION.txt)
+          export CONDA=${HOME}/miniconda
+          export PATH=${CONDA}/bin:$HOME/.local/bin:${PATH}
+          $GITHUB_WORKSPACE/.ci/setup.sh || exit -1
+          $GITHUB_WORKSPACE/.ci/test.sh || exit -1
+  r-check-docs:
+    name: r-package-check-docs
+    timeout-minutes: 120
+    runs-on: ubuntu-latest
+    container: rocker/verse
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2.3.4
+        with:
+          fetch-depth: 5
+          submodules: true
+      - name: Install packages
+        shell: bash
+        run: |
+          Rscript -e "install.packages(c('R6', 'data.table', 'jsonlite', 'roxygen2', 'testthat'), repos = 'https://cran.r-project.org')"
+          sh build-cran-package.sh || exit -1
+          R CMD INSTALL --with-keep.source lightgbm_*.tar.gz || exit -1
+      - name: Test documentation
+        shell: bash --noprofile --norc {0}
```

Comment:
when you just use `shell: bash`, GitHub Actions actually uses `bash --noprofile --norc -e -o pipefail {0}` (where `{0}`  is a placeholder for your code from the `run:` block). This causes this command to fail if no changes are found in the docs:

```shell
num_doc_files_changed=$(
    git diff --name-only | grep --count -E "\.Rd|NAMESPACE"
)
```

If `git diff`  doesn't match anything, the full command returns a non-zero exit code.

You can try this yourself. Run the following in any git repo with no changes (i.e. `git status` says "nothing to commit, working tree clean").

```shell
git diff --name-only | grep --count -E "abc"
echo $?
```

This isn't a problem in the existing r-package-check-docs task on `master` because we run it in a script that does not have `-e` or `-o pipefail` set.


Using `bash --noprofile --norc {0}` ensures that the job only fails when we want it to (by calling `exit` explicitly).

Reply:
Excellent explanation, thank you!